kernel/src/transaction/mod.rs

```diff
@@ -448,6 +586,110 @@ impl Transaction {
         Ok(())
     }
 
+    /// Update deletion vectors for files in the table.
+    ///
+    /// This method can be called multiple times to update deletion vectors for different sets of files.
+    ///
+    /// This method takes a map of file paths to new deletion vector descriptors and an iterator
+    /// of scan file data. It joins the two together internally and will generate appropriate
+    /// remove/add actions on commit to update the deletion vectors.
+    ///
+    /// # Arguments
+    ///
+    /// * `new_dv_descriptors` - A map from data file path (as provided in scan operations) to
+    ///   the new deletion vector descriptor for that file.
+    /// * `existing_data_files` - An iterator over FilteredEngineData from scan metadata. The
+    ///   selected elements of each FilteredEngineData must be a superset of the paths that key
+    ///   `new_dv_descriptors`.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if:
+    /// - A file path in `new_dv_descriptors` is not found in `existing_data_files`
+    ///
+    /// # Examples
+    ///
+    /// ```rust,ignore
+    /// let mut txn = snapshot.clone().transaction(Box::new(FileSystemCommitter::new()))?
+    ///     .with_operation("UPDATE".to_string());
+    ///
+    /// let scan = snapshot.scan_builder().build()?;
+    /// let files: Vec<FilteredEngineData> = scan.scan_metadata(engine)?
+    ///     .collect::<Result<Vec<_>, _>>()?
+    ///     .into_iter()
+    ///     .map(|sm| sm.scan_files)
+    ///     .collect();
+    ///
+    /// // Create map of file paths to new deletion vector descriptors
+    /// let mut dv_map = HashMap::new();
+    /// // ... populate dv_map with file paths and their new DV descriptors ...
+    ///
+    /// txn.update_deletion_vectors(dv_map, files.into_iter())?;
+    /// txn.commit(engine)?;
+    /// ```
+    pub fn update_deletion_vectors(
```

Comment:
can we make this `#[internal_api]` for now, just so we're being clear this might change

Reply:
done.